crates/types/src/partition_table.rs

```diff
@@ -68,13 +93,26 @@ impl FixedConsecutivePartitions {
     }
 }
 
-impl FindPartition for FixedConsecutivePartitions {
+impl Versioned for FixedPartitionTable {
+    fn version(&self) -> Version {
+        self.version()
+    }
+
+    fn increment_version(&mut self) {
+        self.increment_version();
+    }
+}
+
+impl<T> FindPartition for T
+where
+    T: Borrow<FixedPartitionTable>,
+{
```

Comment:
Curious to learn why this implemented over Borrow rather than `impl FindPartition for FixedPartitionTable`.